cf-agent/simulate_mode.c

```diff
@@ -63,8 +63,35 @@ static inline void PrintDelimiter()
 static void ManifestStatInfo(const struct stat *st)
 {
     assert(st != NULL);
+
+    errno = 0;
     struct passwd *owner = getpwuid(st->st_uid);
+    if (owner == NULL)
+    {
+        if (errno == 0)
+        {
+            Log(LOG_LEVEL_ERR, "Failed to get file owner name: non-existent user id %d", st->st_uid);
+        }
+        else 
+        {
+            Log(LOG_LEVEL_ERR, "Failed to get file owner name: %s", GetErrorStr());
+        }
+        return;
+    }
+
     struct group *group = getgrgid(st->st_gid);
+    if (group == NULL)
+    {
+        if (errno == 0)
+        {
+            Log(LOG_LEVEL_ERR, "Failed to get file group name: non-existent group id '%d'", st->st_gid);
```

Comment:
`%d` should be changed to `%ju` ?

Reply:
What is the type of `st->st_gid`?

Reply:
unsigned int

Reply:
Yes, `%ju` and `(uintmax_t)st->st_gid` would be better.